components/grid/virtual-scrolling.md

```diff
@@ -8,281 +8,134 @@ published: True
+## Height
 
-* Pixels&mdash;for example, `Height="480px"`.
-* Percent&mdash;for example, `Height="30%"`. If you set the `Height` parameter to a percentage value, ensure that the wrapper of the Grid has a fixed height set in pixels.
-* Relative CSS units like vh, vmin, and vmax&mdash;for example, `Height="30vh"`.
+Set the Grid `Height` parameter to any [valid `string` CSS value](slug:common-features/dimensions), for example, `px`, `%`, `em`, or `vh`. If the Grid should expand vertically, accoding to the available space, then check the article [Adjust Grid Height to Match the Browser Viewport Height](slug:grid-kb-adjust-height-with-browser).
 
-The tabs below show how to set the `Height` parameter with the different value options.
+Set the `Height` value, so that users can't see the whole [`PageSize` of items](#pagesize) at once. Otherwise, empty row skeletons may display in the Grid while users are not scrolling.
 
-<div class="skip-repl"></div>
-````RAZOR Pixel
-@using Telerik.DataSource
-@using Telerik.DataSource.Extensions
-
-<TelerikGrid OnRead="@OnGridRead"
-             TItem="@Product"
-             ScrollMode="@GridScrollMode.Virtual"
-             Height="480px" RowHeight="60" PageSize="20"
-             Sortable="true" FilterMode="@GridFilterMode.FilterRow">
-    <GridColumns>
-        <GridColumn Field="@nameof(Product.Name)" Title="Product Name" />
-        <GridColumn Field="@nameof(Product.Stock)" />
-    </GridColumns>
-</TelerikGrid>
-
-@code {
-    private List<Product> GridData { get; set; } = new List<Product>();
-
-    private async Task OnGridRead(GridReadEventArgs args)
-    {
-        await Task.Delay(200); // simulate network delay
+## PageSize
 
-        DataSourceResult result = GridData.ToDataSourceResult(args.Request);
-
-        args.Data = result.Data;
-        args.Total = result.Total;
-        args.AggregateResults = result.AggregateResults;
-    }
+Set the Grid `PageSize` parameter to an `int` value. The `PageSize` determines:
 
-    protected override void OnInitialized()
-    {
-        GridData = new List<Product>();
-        var rnd = new Random();
+* How many table rows are populated and rendered at any given time.
+* How many data items are requested from the data source when using the [Grid `OnRead` event to load data on demand](slug:components/grid/manual-operations).
 
-        for (int i = 1; i <= 1000; i++)
-        {
-            GridData.Add(new Product()
-            {
-                Id = i,
-                Name = $"Product {i}",
-                Stock = rnd.Next(0, 100)
-            });
-        }
-    }
+Set the `PageSize` value, so that the rendered table rows do fit in the [Grid height](#height). At least one table row must be completely invisible.Otherwise, empty row skeletons may display in the Grid while users are not scrolling. The exact `PageSize` value allows you to balance between better user experience and data request efficiency:
```

Comment:
```suggestion
Set the `PageSize` value so that the rendered table rows fit in the [Grid height](#height). At least one table row must be completely invisible. Otherwise, empty row skeletons may display in the Grid while users are not scrolling. The exact `PageSize` value allows you to balance between better user experience and data request efficiency:
```